rules/framework.bzl

```diff
@@ -28,21 +32,39 @@ def apple_framework(name, apple_library = apple_library, **kwargs):
     framework_packaging_kwargs = {arg: kwargs.pop(arg) for arg in _APPLE_FRAMEWORK_PACKAGING_KWARGS if arg in kwargs}
     kwargs["enable_framework_vfs"] = kwargs.pop("enable_framework_vfs", True)
 
+    infoplists_by_build_setting = kwargs.pop("infoplists_by_build_setting", {})
+    default_infoplists = kwargs.pop("infoplists", [])
+    infoplists = None
+    if len(infoplists_by_build_setting.values()) > 0 or len(default_infoplists) > 0:
+        infoplists = info_plists_by_setting(
+            name = name,
+            infoplists_by_build_setting = infoplists_by_build_setting,
+            default_infoplists = default_infoplists,
+        )
+
     library = apple_library(name = name, **kwargs)
     apple_framework_packaging(
         name = name,
         framework_name = library.namespace,
+        infoplists = infoplists,
         transitive_deps = library.transitive_deps,
         vfs = library.import_vfsoverlays,
         deps = library.lib_names,
         platforms = library.platforms,
         platform_type = select({
-            "@build_bazel_rules_ios//rules/apple_platform:macos": "macos",
             "@build_bazel_rules_ios//rules/apple_platform:ios": "ios",
+            "@build_bazel_rules_ios//rules/apple_platform:macos": "macos",
             "@build_bazel_rules_ios//rules/apple_platform:tvos": "tvos",
             "@build_bazel_rules_ios//rules/apple_platform:watchos": "watchos",
             "//conditions:default": "",
         }),
+        environment_plist = select({
```

Comment:
This would probably better go as a default on the rule attribute if people need to over-ride it?

Reply:
default doesn't allow `select` method, but we ca use the provided environment_plist from kargs if it exists.